src/components/displayAllComponent.js

```diff
@@ -0,0 +1,82 @@
+import React, { Component } from "react";
+import ChooseFolder from "./chooseFolder/chooseFolder";
+import { Redirect } from "react-router-dom";
+import SvgSettingOptions from "./svgSetting/svgSettingOptions";
+import "../components/checkBoxSelection/checkBoxSelection.scss";
+import { changeObj, optimizeSvg } from "../../src/components/functions";
+import { connect } from "react-redux";
+import PropTypes from "prop-types";
+
+class DisplayAllComponent extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      redirect: false
+    };
+  }
+  getOptimizeSvg = async () => {
+    let originalAndOptimised = {};
+    for (var option of this.props.svgOptions) {
+      changeObj(option, this.props.svgObject.plugins);
+    }
+    for (var setting of this.props.svgSettingList) {
+      if (this.props.svgOptions.length > 0) {
+        originalAndOptimised = {
+          originalSvg: setting.dataurl,
+          name: setting.name,
+          optimisedSvg: await optimizeSvg(
+            setting.dataurl,
+            this.props.svgObject.plugins
+          )
+        };
+      } else {
+        originalAndOptimised = {
+          originalSvg: setting.dataurl,
+          name: setting.name,
+          optimisedSvg: await optimizeSvg(
+            setting.dataurl,
+            this.props.svgObject.plugins
+          )
+        };
+      }
+      this.props.displayOptimize.push(originalAndOptimised);
```

Comment:
you should not push directly to a prop, mutability bad

Reply:
yhea, I will assigned a full function instead into the prop but when we tried doing that earlier the elements inside the prop where not reflecting until we decided to push one time.

Reply:
We tried in the beginning to use concat  and other options but items wwere not reflecting.

Reply:
Okay, I have figure away to solve this and have pushed.